graalwasm/graalwasm-embed-c-code-guide/README.md

```diff
@@ -0,0 +1,196 @@
+
+The example below demonstrates how to compile a C function to WebAssembly and run it embedded in a Java application.
+
+## 1. Getting Started
+
+To complete this guide, you need the following:
+- [GraalVM JDK](https://www.graalvm.org/downloads/)
+- [Emscripten compiler frontend](https://emscripten.org/docs/tools_reference/emcc.html)
+- [Maven](https://maven.apache.org/)
+
+## 2. Setting up the Maven Project
+
+To follow this guide, generate the application from the [Maven Quickstart Archetype](https://maven.apache.org/archetypes/maven-archetype-quickstart/):
+
+```shell
+mvn archetype:generate -DarchetypeGroupId=org.apache.maven.archetypes -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.5 -DgroupId=com.example -DartifactId=demo -DinteractiveMode=false
+cd demo
+```
+
+### 2.1. Adding the Polyglot API and GraalWasm Dependencies
+
+The GraalVM SDK Polyglot API can be easily added as a Maven dependency to your Java project.
+The GraalWasm artifact should be on the Java module or class path too.
+
+Add the following set of dependencies to the `<dependencies>` section of your project's _pom.xml_:
+
+   - To add the Polyglot API:
+       ```xml
+       <!-- <dependencies> -->
+       <dependency>
+           <groupId>org.graalvm.polyglot</groupId>
+           <artifactId>polyglot</artifactId>
+           <version>24.1.0</version>
+       </dependency>
+       <!-- </dependencies> -->
+       ```
+   - To add GraalWasm:
+       ```xml
+       <!-- <dependencies> -->
+       <dependency>
+           <groupId>org.graalvm.polyglot</groupId>
+           <artifactId>wasm</artifactId>
+           <version>24.1.0</version>
+           <type>pom</type>
+       </dependency>
+       <!-- </dependencies> -->
+       ```
+
+## 3. Setting Up C Code
+
+Next, we will write a C function and compile it into a WebAssembly module.
+
+### 3.1 Writing C Code
+
+Put the following C program in _src/main/c/floyd.c_:
+
+```c
+#include <stdio.h>
+
+void floyd() {
+    int number = 1;
+    int rows = 10;
+    for (int i = 1; i <= rows; i++) {
+        for (int j = 1; j <= i; j++) {
+            printf("%d ", number);
+            ++number;
+        }
+        printf(".\n");
+    }
+}
+
+int main() {
+    floyd();
+    return 0;
+}
+```
+
+Note that `floyd` is defined as a separate function and can be exported.
+
+### 3.2 Compiling C Code to WebAssembly
+
+Compile the C code using the most recent version of the [Emscripten compiler frontend](https://emscripten.org/docs/tools_reference/emcc.html):
+
+```shell
+mkdir -p target/classes/com/example
+emcc --no-entry -s EXPORTED_FUNCTIONS=_floyd -o target/classes/com/example/floyd.wasm src/main/c/floyd.c
+```
+
+> The exported functions must be prefixed by `_`. If you reference that function in the Java code, the exported name should not contain the underscore.
+
+It produces a standalone file _floyd.wasm_ in _target/classes/com/example/_, which lets us load the file as a resource.
```

Comment:
which lets us load the file as a resource.  >  which enables you to load the file as a resource.